pkg/controller/s3/bucket/accelerateConfig.go

```diff
@@ -68,7 +68,7 @@ func NewAccelerateConfigurationClient(client s3.BucketClient) *AccelerateConfigu
 // Observe checks if the resource exists and if it matches the local configuration
 func (in *AccelerateConfigurationClient) Observe(ctx context.Context, bucket *v1beta1.Bucket) (ResourceStatus, error) {
 	external, err := in.client.GetBucketAccelerateConfigurationRequest(&awss3.GetBucketAccelerateConfigurationInput{Bucket: aws.String(meta.GetExternalName(bucket))}).Send(ctx)
-	if err != nil && !s3.AllowedParameters(err) {
+	if err != nil && !s3.MethodNotSupported(err) && !s3.ArgumentNotSupported(err) {
 		return NeedsUpdate, errors.Wrap(err, accelGetFailed)
 	}
 	if bucket.Spec.ForProvider.AccelerateConfiguration != nil &&
```

Comment:
Would `external.Status` result in `nil` pointer exception if one of those errors are returned? I vaguely remember that even in error case, this API returns an object that's not `nil` but I'm not sure whether that behavior is consistent between different AWS endpoints.

Reply:
@Promaethius just pinging to check whether you had the chance to validate that. FWIW, you can also just:
```
external, err := in.client.GetBucketAccelerateConfigurationRequest(&awss3.GetBucketAccelerateConfigurationInput{Bucket: aws.String(meta.GetExternalName(bucket))}).Send(ctx)
if s3.MethodNotSupported(err) || s3.ArgumentNotSupported(err){
  return Updated, nil
}
if err != nil {
...
```
to make sure it doesn't proceed if the error is of those kinds.